### DISCUSSION TITLE
Screens & Roles

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#4235] [Help] [answered]
Hello 😄,

I created about 9 Roles and screens for these roles so you can only access the f.e. "Bios" screen with the "Bios" role.
The thing is, when I'm an Admin and I click on one of these screens, it redirects to a different screen with white screen.
So I guess it works, because the screen I get redirected to, is white which means I can't see the Table because I don't have the role, but why does it redirect to a random screen? Is that a bug or what do I do?

Thanks in advance 😃

**Comment by mjashanks** [comment, marked as answer]
Hi There.

I would expect an admin to have access to all of the screens - as it would be the highest user level. Is Admin the highest user level? If not, then Admin may not actually have access to this screen.

If that is the case, then I'm not surprised that they get redirected to a random screen. It would be nicer to have a default 404 page, but a random page is the fallback for now!

If the **should** have access, then I'd be interested to know more about your app. If possible, I'd like to see an export - feel free to send to support@budibase.com if you'd like me to debug it!

**Reply by NicolasFabre81** [reply]
Hello @mjashanks , is the random page still the fallback now ? Or can we configure a default error page...

**Comment by kasgithub8** [comment]
Hey @mjashanks,

I sent the App to support@budibase.com!

**Reply by mjashanks** [reply]
Thanks for this! I see your issue. I have logged this as an issue - a bug with custom roles: https://github.com/Budibase/budibase/issues/4267

We'll take a look at this. Thanks!

**Comment by kasgithub8** [comment]
Thank you very much! This would help me a lot!
